lib/everypolitician/popolo/event.rb

```diff
@@ -38,8 +38,29 @@ def wikidata
       attr_reader :document
     end
 
+    class DynamicEventClassFinder
+      def self.new(doc, *args)
+        case doc[:classification]
```

Comment:
We've only just removed one giant `case` statement for this sort of thing. Bringing back another one doesn't seem ideal. We can't quite do the same declarative in-place approach as then, but making this data driven seems like it might be slightly better.